test/cri-containerd/sandbox.go

```diff
@@ -35,28 +62,38 @@ func removePodSandbox(t *testing.T, client runtime.RuntimeServiceClient, ctx con
 	}
 }
 
-func getRunPodSandboxRequest(t *testing.T, runtimeHandler string, annotations map[string]string) *runtime.RunPodSandboxRequest {
-	req := &runtime.RunPodSandboxRequest{
-		Config: &runtime.PodSandboxConfig{
-			Metadata: &runtime.PodSandboxMetadata{
-				Name:      t.Name(),
-				Namespace: testNamespace,
-			},
-			Annotations: annotations,
+func getTestSandboxConfig(t *testing.T, opts ...SandboxConfigOpt) *runtime.PodSandboxConfig {
+	c := &runtime.PodSandboxConfig{
+		Metadata: &runtime.PodSandboxMetadata{
+			Name:      t.Name(),
+			Namespace: testNamespace,
 		},
-		RuntimeHandler: runtimeHandler,
 	}
 
 	if *flagVirtstack != "" {
-		if req.Config.Annotations == nil {
-			req.Config.Annotations = make(map[string]string)
-		}
-		req.Config.Annotations["io.microsoft.virtualmachine.vmsource"] = *flagVirtstack
-		req.Config.Annotations["io.microsoft.virtualmachine.vmservice.address"] = testVMServiceAddress
-		req.Config.Annotations["io.microsoft.virtualmachine.vmservice.path"] = testVMServiceBinary
+		opts = append(opts, WithSandboxAnnotations(map[string]string{
+			"io.microsoft.virtualmachine.vmsource":          *flagVirtstack,
+			"io.microsoft.virtualmachine.vmservice.address": testVMServiceAddress,
+			"io.microsoft.virtualmachine.vmservice.path":    testVMServiceBinary,
+		}))
 		if *flagVMServiceBinary != "" {
-			req.Config.Annotations["io.microsoft.virtualmachine.vmservice.path"] = *flagVMServiceBinary
+			opts = append(opts, WithSandboxAnnotations(map[string]string{
+				"io.microsoft.virtualmachine.vmservice.path": *flagVMServiceBinary,
+			}))
+		}
+	}
+
+	for _, o := range opts {
+		if err := o(c); err != nil {
+			t.Fatalf("failed to apply PodSandboxConfig option: %s", err)
```

Comment:
Should we do these first so that if flagVirtstack or flagVMServiceBinary get priority in annotation value?

Reply:
I see what you mean. I think I'll just have a local variable that will keep the value of `"io.microsoft.virtualmachine.vmservice.path"` with `testVMServiceBinary` being the default one.

Reply:
I'm not sure I get what you mean by that?

Reply:
ok, I must have misunderstood. the use case you're thinking about is when the vm service is overriden via opts passed explicitly?

Reply:
discussed offline.